TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/DriveSystem.java

```diff
@@ -254,28 +301,33 @@ public boolean turn(double degrees, double maxPower) {
      * @param speed     Desired speed of turn
      */
     public boolean onHeading(double speed, double heading) {
-        double steer;
         double leftSpeed;
-        double rightSpeed;
 
         // determine turn power based on +/- error
         double error = getError(heading);
 
+        // If it gets there: stop
         if (Math.abs(error) <= HEADING_THRESHOLD) {
             mTargetHeading = 0;
             setMotorPower(0);
+            mTurnCounter = 0;
             return true;
         }
 
-        steer = getSteer(error);
-        leftSpeed  = speed * steer;
-        rightSpeed   = -leftSpeed;
+        // TODO
+        // Go full speed until 60% there
+        leftSpeed = error > Math.abs(0.85 * (mInitHeading)) ? speed : (speed * getSteer(error));
```

Comment:
And what exactly are you doing here? What's with `mInitHeading`?

Reply:
What's wrong with mInitHeading?

Reply:
Nothing is inherently wrong with it. Just unclear on why you're using it.

Reply:
It's the heading at the start of the turn, so that we can see if we made it 15 % of the way there before starting ramping.

Reply:
But isn't that essentially what `error` is doing? Couldn't you just say "if `error` > (some percentage)..."